backend/infrahub/core/node/__init__.py

```diff
@@ -581,12 +587,22 @@ async def _update(
                     node_changelog.add_attribute(attribute=updated_attribute)
 
         # Go over the list of relationships and update them one by one
+        processed_relationships: list[str] = []
         for name in self._relationships:
             if (fields and name in fields) or not fields:
+                processed_relationships.append(name)
                 rel: RelationshipManager = getattr(self, name)
                 updated_relationship = await rel.save(at=update_at, db=db)
                 node_changelog.add_relationship(relationship=updated_relationship)
 
+        if len(processed_relationships) != len(self._relationships):
+            # Analyze if the node has a parent and add it to the changelog if missing
+            if parent_relationship := self._get_parent_relationship_name():
+                if parent_relationship not in processed_relationships:
+                    rel: RelationshipManager = getattr(self, parent_relationship)
+                    for peer in await rel.get_relationships(db=db):
```

Comment:
Couldn't a node have only a single parent? In which case a `rel.get_parent` method seems approriate

Reply:
We are always calling `.get_relationships()` in the RelationshipManager even for cardinality=one relationships. But I can add a rel.get_parent() for clarity.